packages/hash/frontend/src/components/hooks/usePageComments.ts

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from "@apollo/client";
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import {
+  GetPersistedPageCommentsQuery,
+  GetPersistedPageCommentsQueryVariables,
+} from "../../graphql/apiTypes.gen";
+import { getPersistedPageComments } from "../../graphql/queries/page.queries";
+
+export type PageThread = PageComment & {
+  replies: PageComment[];
+};
+
+export type PageComment = {
+  ownedById: string;
+  entityId: string;
+  hasText: Array<TextToken>;
+  textUpdatedAt: string;
+  author: { entityId: string; properties: any };
+  parent: { entityId: string };
+};
+
+export type PageCommentsInfo = {
+  data: PageThread[];
+  loading: boolean;
+};
+
+export const usePageComments = (pageId: string): PageCommentsInfo => {
+  const { data, loading } = useQuery<
+    GetPersistedPageCommentsQuery,
+    GetPersistedPageCommentsQueryVariables
+  >(getPersistedPageComments, {
+    variables: { entityId: pageId },
+  });
+
+  return { data: data?.persistedPageComments ?? [], loading };
```

Comment:
You probably want to save this empty array to a const – or any downstream code that uses this in an effect or whatever will fire on every render.

Reply:
Is this the right way to do this? 5f9e9b26d8f708f95ac3f01a8946eb815c7f4d4a

Reply:
Yup!